cron/hourly/70-cancel-subscriptions-for-cancelled-orders.ts

```diff
@@ -28,24 +29,38 @@ const getHostFromOrder = async order => {
   return models.Collective.findByPk(transactions[0].HostCollectiveId);
 };
 
+const getOrderCancelationReason = (collective, order, orderHost) => {
+  if (order.TierId && !order.Tier) {
+    return ['DELETED_TIER', `Order tier deleted`];
+  } else if (collective.deactivatedAt) {
+    return ['ARCHIVED_ACCOUNT', `@${collective.slug} archived their account`];
+  } else if (!collective.HostCollectiveId) {
+    return ['UNHOSTED_COLLECTIVE', `@${collective.slug} was un-hosted`];
+  } else if (collective.HostCollectiveId !== orderHost.id) {
+    return ['CHANGED_HOST', `@${collective.slug} changed host`];
+  } else {
+    return ['CANCELLED_ORDER', `Order cancelled`];
+  }
+};
+
 /**
  * When archiving collectives. we need to make sure subscriptions managed externally are properly cancelled
  * by calling the right service method (PayPal). We do this asynchronously to properly deal with rate limiting and
  * performance constraints.
  */
 export async function run() {
   const orphanOrders = await models.Order.findAll({
+    where: { status: OrderStatuses.CANCELLED },
     include: [
       {
-        model: models.Subscription,
-        required: true,
-        where: { isManagedExternally: true, isActive: true },
+        model: models.Tier,
       },
       {
-        association: 'collective',
+        model: models.Subscription,
         required: true,
-        where: { deactivatedAt: { [Op.not]: null }, isActive: false },
+        where: { isActive: true },
```

Comment:
The following query returns 5 results in prod:

```sql
select * from "Orders" o
INNER JOIN "Subscriptions" s on o."SubscriptionId" = s.id
where o.status = 'CANCELLED'
AND s."isActive"
```

They're old, so I'm assuming it's the leftover of a past bug. I'm noting their IDs just in case and manually marking the subscriptions as inactive:

> 
> 5607
> 525029
> 505772
> 174267
> 181094